core/pkg/opencost/cloudcostprops.go

```diff
@@ -283,3 +288,26 @@ func (ccp *CloudCostProperties) GenerateKey(props []string) string {
 
 	return strings.Join(values, "/")
 }
+
+// HashKey creates a key on the entire property set including labels of a uniform length.
+// This key is meant to be used when constructing unaggregated CloudCostSet for storage
+func (ccp *CloudCostProperties) hashKey() string {
+	builder := strings.Builder{}
+	builder.WriteString(ccp.ProviderID)
+	builder.WriteString(ccp.Provider)
+	builder.WriteString(ccp.AccountID)
+	builder.WriteString(ccp.InvoiceEntityID)
+	builder.WriteString(ccp.Service)
+	builder.WriteString(ccp.Category)
+
+	// Sort label keys before adding key/value pairs to the hash string
+	labelKeys := maps.Keys(ccp.Labels)
+	sort.Strings(labelKeys)
+	for _, k := range labelKeys {
+		builder.WriteString(k)
+		builder.WriteString(ccp.Labels[k])
+	}
```

Comment:
Would you please add a comment explaining the bit about label erasure that motivates this code? You explained it well in the PR description